apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -782,6 +788,18 @@ public void activate(ElasticContext<?> context) {
             logger.debug("Activating {} on thread {}", context, Thread.currentThread().getId());
         }
 
+        if (activeStack.get().size() == coreConfiguration.getTransactionMaxSpans()) {
```

Comment:
As this issue can also happen with a misconfigured trace_methods setting, I wonder if we should also explicitly check whether we've already reached the span limit in the trace methods advice before creating a span.

Reply:
Good idea! I would say even doing so when the number of spas reaches 90-95% of max-spans, so to allow room for the really important spans.

Reply:
Misconfiguration of `trace_methods` is indeed a worthy issue to tackle, but I prefer to leave it out of this PR.
The current fix is dealing more with activation leak symptoms, where the activation stack increases to really monstrous size, whereas in common misconfiguration of `trace_methods`, the number of active spans should still be limited by thread stacks depths, so the JVM is likely to get suffocated by other parameters.
We currently issue a warning once per 5 minutes when the span limit is reached, normally due to this improper configuration.

Reply:
I agree that this should be handled in a separate PR.

Maybe something else we could think about: issue a warning or even clear the stack if the previously active span has a different trace id. Could be expensive though.